### DISCUSSION TITLE
Problem with implementing MUMPS in Windows build

### DISCUSSION THREAD

**Discussion by hoodradd** [OpenRadioss/OpenRadioss#3069] [Q&A]
Hello, i were trying to implement the MUMPS library in Windows building for using the impicit solver, but after redoing the cmakelists file, i were stucked getting the same error. Maybe someone else tried to do it, and know how to fix this. It says that i have unresolved external symbols __imp_strdup, __imp__strdip and __imp_ferror
![image](https://github.com/user-attachments/assets/2704ac9f-1dbd-4272-8055-67ac0e365e3a)

**Comment by OlivierWienholtz** [comment]
Hi Hoodradd

What compiler did you use ?
On Windows strdup  call is deprecated and has to be replaced by call to _strdup
I checked the culprit is not coming from the Engine (we habe specific code do distinguish Linux & Windows code.

The other ones are more strange : 
ferror() call exists in Windows & is not deprecated.
strdip : did you mean _strdup

Best Regards
Olivier

**Reply by laurent-altr** [reply]
Hi @hoodradd, 

I added a new way to use MUMPS on windows with intel compiler: you need to download  [MUMPS](http://ftp.mcs.anl.gov/pub/petsc/externalpackages/MUMPS_5.5.1.tar.gz) and extract it into the ./engine/extlib folder, such that ./engine/extlib/MUMPS_5.5.1 exists. Then you can build OpenRadioss as usual.

Thank you, 
Laurent
